parser/ParseBlockProperties.hpp

```diff
@@ -0,0 +1,245 @@
+/**
+ *   Copyright 2016 Pivotal Software, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_PARSER_PARSE_BLOCK_PROPERTIES_HPP_
+#define QUICKSTEP_PARSER_PARSE_BLOCK_PROPERTIES_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "parser/ParseString.hpp"
+#include "parser/ParseTreeNode.hpp"
+#include "utility/Macros.hpp"
+#include "utility/PtrList.hpp"
+
+namespace quickstep {
+
+/** \addtogroup Parser
+ *  @{
+ */
+
+/**
+ * @brief Parsed BlockPropertyItem which simply a key (property) and the 
+ *        corresponding value.
+ * @details The value may actually be a list of Parse strings. This is to cover 
+ *          the case where a user specifies a subset of columns to compress 
+ *          rather than using the keyword 'ALL'. 
+ */
+class ParseBlockPropertyItem : public ParseTreeNode {
+ public:
+  enum class Property {
```

Comment:
This doesn't handle describing IndexSubBlocks (although I understand another student is working on that)... what about the number of slots for a StorageBlockLayout, though? How would the user specify that they want more than 1 slot?